src/App.js

```diff
@@ -63,73 +63,187 @@ const useStyles = makeStyles({
-  const [prevVal, setPrevVal] = useState(null);
-  const [equation, setEquation] = useState([]);
+  const [currentToken, setCurrentToken] = useState('0');
+  const [prevInput, setPrevInput] = useState('');
+  const [equation, setEquation] = useState('');
+  const [answer, setAnswer] = useState(null);
 
-  const handleClick = (e) => {
-    const input = e.target.innerHTML;
-    const isOperand = input.match(/[0-9]/);
-    const isOperator = input.match(/[/+\-X]/);
-    const isEquals = input.match(/=/);
-    const isDecimal = input.match(/./);
-    const isClear = input.match(/AC/);
-    const startsWithAngle = input.match(/</);
-    if (startsWithAngle) {
-      return;
+  const endsWithOperand = /\d$/;
+  const endsWithOperator = /[+-/*]$/;
+  const isDecimal = /^\d*\.?\d*$/;
+  const endsWithNegative = /-$/;
+  const isNegative = /-/;
+  const endsWithNonNegativeOperator = /[*+/]$/;
+  const isOperator = /[*/+-]/;
+  const isMultiply = /X/;
+  const isOperand = /\d/;
+  const isNonNegOperator = /[*+/]/;
+
+  const handleOperator = ({ currentTarget: { value } }) => {
+    let input;
+    if (isMultiply.test(value)) {
+      input = value.replace('X', '*');
+    } else {
+      input = value;
     }
-    if (input === '0' && equation[0] === '0') {
-      return;
+    if (endsWithOperand.test(equation) && !equation.includes('=')) {
+      console.log('endWithOperand, conditon 1');
+      setCurrentToken(input);
+      setEquation(equation + input);
+    } else if (
+      isOperand.test(equation.slice(-2, -1)) &&
+      endsWithOperator.test(equation) &&
+      isNonNegOperator.test(input)
+    ) {
+      console.log('condition 2');
+      console.log('equation:', equation);
+      console.log('2nd to last item in equation:', equation.slice(-2, -1));
+      console.log(
+        'endsWIthNonNegOperator:',
+        endsWithNonNegativeOperator.test(equation)
+      );
+      setCurrentToken(input);
+      setEquation(equation.slice(0, -1) + input);
+    } else if (
+      isOperand.test(equation.slice(-2, -1)) &&
+      endsWithOperator.test(equation) &&
+      isNegative.test(input)
+    ) {
+      console.log('condition 3');
+      setEquation(equation + input);
+      setCurrentToken(input);
+    } else if (
+      isOperator.test(equation.slice(-2, -1)) &&
+      endsWithNegative.test(equation) &&
+      isNonNegOperator.test(input)
+    ) {
+      console.log('condition 4');
+      setEquation(equation.slice(0, -2) + input);
+      setCurrentToken(input);
+    } else if (
+      isOperator.test(currentToken) &&
+      isOperator.test(equation.slice(-1))
+    ) {
+      console.log('condition 5 operator');
+      setCurrentToken(input);
+      setEquation(equation.slice(0, -1) + input);
+    } else if (equation === '') {
+      setCurrentToken(input);
+      setEquation(input);
+    } else if (equation.includes('=')) {
+      console.log('handleOperator condition 6');
+      setCurrentToken(input);
+      setEquation(answer + input);
```

Comment:
With the long `else if` chain it is hard to read at a glance. What do you think about defining a function for each of the test conditions with a descriptive name? The idea would be something that reads like:

```javascript
// Example of helper function for use in test conditions in `handleOperator`:
const function addOperator() { 
  return endsWithOperand.test(equation) && !equation.includes('=')
}

...

// Inside the handleOperator function:
if (addOperator()) {
  setCurrentToken(input);
  setEquation(equation + input);
  // return early
  return 
}
.
.
.
// Each test condition can be its own if statement.
if (...) {
...
}
// end of handleOperator function
```